covariance/covarianceBase.cpp

```diff
@@ -904,21 +888,24 @@ void covarianceBase::SetBranches(TTree &tree, bool SaveProposal) {
 // ********************************************
 void covarianceBase::setStepScale(const double scale) {
 // ********************************************
-  if(scale == 0)
-  {
+  if(scale == 0) {
     MACH3LOG_ERROR("You are trying so set StepScale to 0 this will not work");
     throw MaCh3Exception(__FILE__ , __LINE__ );
   }
   MACH3LOG_INFO("{} setStepScale() = {}", getName(), scale);
+  const double SuggestedScale = 2.38*2.38/_fNumPar;
+  if(std::fabs(scale - SuggestedScale)/SuggestedScale > 1) {
```

Comment:
This won't warn you if you go too small right? ie scale=0 won't throw an error. Maybe make it >0.8?

Reply:
Case scale = 0 is handle separately just above this check. Atlhough we don't have any checks for negative so will update it